src/backends/torch/torchdataset.cc

```diff
@@ -217,7 +217,8 @@ namespace dd
       {
         std::vector<uint8_t> img_target_data(targets.begin(), targets.end());
         bw_target = cv::Mat(img_target_data, true);
-        bw_target = cv::imdecode(bw_target, CV_LOAD_IMAGE_GRAYSCALE);
+        bw_target = cv::imdecode(
+            bw_target, CV_LOAD_IMAGE_UNCHANGED); // CV_LOAD_IMAGE_GRAYSCALE);
```

Comment:
forgot to change back to GRAYSCALE?

Reply:
Yes, this is a good catch, as usual, thanks ! At the moment it makes no difference if the target files are properly built, but loading in grayscale is more robust anyways.